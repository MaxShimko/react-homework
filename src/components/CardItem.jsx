import React from 'react';

import '../styles/CardItem.scss';

export default function CardItem({title, ganre, posterUrl, year}) {
    return (
        <div className="card-item">
            <img src={posterUrl} />
            <div className="delimiter-1" />
            <div className="description">
                <div className="left">
                    <h2>{title}</h2>
                    <p>{ganre}</p>
                </div>
                <div className="right">
                    <div className="year">{year}</div>
                </div>
            </div>
        </div>
    );
};

