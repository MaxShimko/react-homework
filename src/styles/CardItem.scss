.card-item {
  padding: 20px;
  cursor: pointer;

  img {
    width: 100%;
  }

  .description {
    display: flex;
    justify-content: space-between;
    color: rgba(#fff, 0.7);

    .left {
      padding-right: 10px;
      h2 {
        font-size: 16px;
        margin: 0;
      }

      p {
        margin: 0;
        font-size: 12px;
        font-family: "Montserrat-Light", sans-serif;
      }
    }

    .right {
      .year {
        font-size: 12px;
        border-radius: 4px;
        border: 1px solid rgba(#fff, 0.2);
        padding: 0 5px;
      }
    }
  }
}
