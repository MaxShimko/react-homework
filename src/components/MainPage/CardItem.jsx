import React from 'react';

import '../../styles/CardItem.scss';

class CardItem extends React.PureComponent {
    render() {
        return <div className="card-item">
            <img src={this.props.posterUrl} />
            <div className="delimiter-1"></div>
            <div className="description">
                <div className="left">
                    <h2>{this.props.title}</h2>
                    <p>{this.props.ganre}</p>
                </div>
                <div className="right">
                    <div className="year">{this.props.year}</div>
                </div>
            </div>
        </div>;
    }
}

export default CardItem;
