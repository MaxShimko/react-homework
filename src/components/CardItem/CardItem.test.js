import React from 'react';
import { shallow } from 'enzyme';
import CardItem from './CardItem';
import {describe, expect, it} from '@jest/globals';
import toJson from 'enzyme-to-json';

describe('<CardItem />', () => {
    it('renders without crashing', () => {
        const component = shallow(<CardItem />);
        expect(toJson(component)).toMatchSnapshot();
    });
});
