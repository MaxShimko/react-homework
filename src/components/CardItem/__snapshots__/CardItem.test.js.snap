// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardItem /> renders without crashing 1`] = `
<div
  className="card-item"
>
  <img />
  <div
    className="delimiter-1"
  />
  <div
    className="description"
  >
    <div
      className="left"
    >
      <h2 />
      <p />
    </div>
    <div
      className="right"
    >
      <div
        className="year"
      />
    </div>
  </div>
</div>
`;
